src/EditorFeatures/Test/EmbeddedLanguages/StackFrame/StackFrameSyntaxFactory.cs

```diff
@@ -0,0 +1,204 @@
+
+        public static StackFrameParameterList ParameterList(params StackFrameParameterDeclarationNode[] parameters)
+            => ParameterList(OpenParenToken, CloseParenToken, parameters);
+
+        public static StackFrameParameterList ParameterList(StackFrameToken openToken, StackFrameToken closeToken, params StackFrameParameterDeclarationNode[] parameters)
+        {
+            var separatedList = parameters.Length == 0
+                ? SeparatedStackFrameNodeList<StackFrameParameterDeclarationNode>.Empty
+                : new(CommaSeparateList(parameters));
+
+            return new(openToken, separatedList, closeToken);
+
+            static ImmutableArray<StackFrameNodeOrToken> CommaSeparateList(StackFrameParameterDeclarationNode[] parameters)
+            {
+                var builder = ImmutableArray.CreateBuilder<StackFrameNodeOrToken>();
+                builder.Add(parameters[0]);
+
+                for (var i = 1; i < parameters.Length; i++)
+                {
+                    builder.Add(CommaToken);
+                    builder.Add(parameters[i]);
+                }
+
+                return builder.ToImmutable();
+            }
+        }
+
+        public static StackFrameMethodDeclarationNode MethodDeclaration(
+            StackFrameQualifiedNameNode memberAccessExpression,
+            StackFrameTypeArgumentList? typeArguments = null,
+            StackFrameParameterList? argumentList = null)
+        {
+            return new StackFrameMethodDeclarationNode(memberAccessExpression, typeArguments, argumentList ?? ParameterList(OpenParenToken, CloseParenToken));
+        }
+
+        public static StackFrameQualifiedNameNode QualifiedName(string s, StackFrameTrivia? leadingTrivia = null, StackFrameTrivia? trailingTrivia = null)
+            => QualifiedName(s, CreateTriviaArray(leadingTrivia), CreateTriviaArray(trailingTrivia));
+
+        public static StackFrameQualifiedNameNode QualifiedName(string s, ImmutableArray<StackFrameTrivia> leadingTrivia, ImmutableArray<StackFrameTrivia> trailingTrivia)
+        {
+            StackFrameNameNode? current = null;
+            var identifiers = s.Split('.');
+            for (var i = 0; i < identifiers.Length; i++)
+            {
+                var identifier = identifiers[i];
+
+                if (current is null)
+                {
+                    current = Identifier(IdentifierToken(identifier, leadingTrivia: leadingTrivia, trailingTrivia: ImmutableArray<StackFrameTrivia>.Empty));
+                }
+                else if (i == identifiers.Length - 1)
+                {
+                    var rhs = Identifier(IdentifierToken(identifier, leadingTrivia: ImmutableArray<StackFrameTrivia>.Empty, trailingTrivia: trailingTrivia));
+                    current = QualifiedName(current, rhs);
+                }
+                else
+                {
+                    current = QualifiedName(current, Identifier(identifier));
+                }
+            }
```

Comment:
add test for strange things like `X.Y.M.(...)`

Reply:
Added to `TestInvalidInputs`